var/spack/repos/builtin/packages/xnedit/package.py

```diff
@@ -0,0 +1,58 @@
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Xnedit(MakefilePackage):
+    """XNEdit is a multi-purpose text editor for the X Window System, which
+    combines a standard, easy to use, graphical user interface with the thorough
+    functionality and stability required by users who edit text eight hours a
+    day. It provides intensive support for development in a wide variety of
+    languages, text processors, and other tools, but at the same time can be
+    used productively by just about anyone who needs to edit text.
+
+    XNEdit is a fork of the Nirvana Editor (NEdit) and provides new
+    functionality like antialiased text rendering and support for unicode.
+    """
+
+    homepage = "https://www.unixwork.de/xnedit/"
+    url = "https://github.com/unixwork/xnedit/archive/refs/tags/v1.5.2.tar.gz"
+    git = "https://github.com/unixwork/xnedit.git"
+
+    maintainers("davekeeshan")
+
+    version("1.5.2", sha256="2f9710b661f8ec5d371a3385fa480c7424e2f863938a8e2ae71cb17397be3f91")
+    version("1.5.1", sha256="c871589e912ed9f9a02cc57932f5bb9694ec91cc5487be0cd55e7d3aade372d6")
+    version("1.5.0", sha256="054f0805405bfc0304e5d27aaac7f4d89b9f028b9e408bf8c079669ddb89df00")
+    version("1.4.1", sha256="6f6ef827b0f6efe481333fe81ec30e2ab133b24cbc9bab224c02e1f69474bade")
+    version("1.4.0", sha256="91c88689c853a6f16f22c109a7283ae552a7079829a25dfa6db192b3d6c6cb60")
+    version("1.3.3", sha256="388765726bca2887eca4a95e1253214ee7209075b0bdd4a143ad870ccc04703f")
+    version("1.3.2", sha256="20f816dba7c3aa59504d1a1360a731854581801c3031012a6b8fef55d437bd04")
+    version("1.3.1", sha256="e1551983cb4e6ee8c9dcb35a628fdcf12dd191bab9321c98c77c22456688e81a")
+    version("1.3.0", sha256="5b1fccbaee007dceca37b24d248e27818c30eb0a87b79317f51ba731da3b8a03")
+    version("1.2.2", sha256="9d0434fb47a306f8665bcf4f9bf3f41173c31010cdfd767a0622ae8b9be10ac1")
+    version("1.2.1", sha256="e0c0147c7f81ccac1c40acb5636b375ef76678a7a84ec592d509461af00983c5")
+    version("1.2.0", sha256="8a6a1252b36be8be5c20aafe5989ad3158cd66449632bf1014ad10cf85a6ea16")
+    version("1.1.1", sha256="9958cc58f9834e3ccfa1d094e942f1ff5bc8b0464ae8e6d259cbb9e71bfb1ebd")
+    version("1.1.0", sha256="c335c3a5beb2263d5a9974931dbdd1f2e5cf77fd0eafd4026ee2b7ca0aad2fe7")
+    version("1.0.1", sha256="3efa26d180696ea7b24c3efd2599c52183b6851fc1bc87ce9a4f85d465962a8c")
+    version("1.0.0", sha256="f58dcbd268f226192584f56dd1a897290a66176d91a90d715a40d63578a84b72")
+
+    depends_on("automake", type="build")
+    depends_on("libx11", type="build")
+    depends_on("libxt", type="build")
+    depends_on("libxpm", type="build")
+    depends_on("motif", type="build")
+    depends_on("pcre", type="build")
+
+    def setup_build_environment(self, env):
+        env.prepend_path("LDFLAGS", "-Wl,--copy-dt-needed-entries")
```

Comment:
If you make the above `link` dependencies, you probably don't need this?

Reply:
I had actually being doing that, but it fails to build:

```
     95     cc -O -std=gnu99 -I/usr/X11R6/include -I/usr/include/X11 -DUSE_LPR_PRINT_CMD -I/cadtools/linux-ubuntu22.04-skylake/spack/opt
            /spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libx11/1.8.4/include -I/cadtools/linux-ubuntu22.04-skylake/spack/opt/spack/linux
            -ubuntu22.04-skylake/gcc-12.3.0/libxcb/1.14/include -I/cadtools/linux-ubuntu22.04-skylake/spack/opt/spack/linux-ubuntu22.04-
            skylake/gcc-12.3.0/libxdmcp/1.1.4/include -I/cadtools/linux-ubuntu22.04-skylake/spack/opt/spack/linux-ubuntu22.04-skylake/gc
            c-12.3.0/libxau/1.0.8/include -I/cadtools/linux-ubuntu22.04-skylake/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/uti
            l-linux-uuid/2.38.1/include/uuid -I/cadtools/linux-ubuntu22.04-skylake/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/
            zlib-ng/2.1.4/include -I/cadtools/linux-ubuntu22.04-skylake/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/xproto/7.0.
            31/include -I/cadtools/linux-ubuntu22.04-skylake/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/kbproto/1.0.7/include 
            -I/usr/include/freetype2 -I/usr/include/libpng16 nedit.o file.o menu.o window.o selection.o search.o undo.o shift.o help.o p
            references.o tags.o userCmds.o shell.o regularExp.o macro.o text.o textSel.o textDisp.o textBuf.o textDrag.o server.o highli
            ght.o highlightData.o interpret.o parse.o smartIndent.o regexConvert.o windowTitle.o calltips.o server_common.o rangeset.o e
            ditorconfig.o linkdate.o \
     96      ../util/libNUtil.a  ../Microline/XmL/libXmL.a ../Xlt/libXlt.a -L/usr/X11R6/lib -lpcre -lXm -lXt -lX11 -lXrender -lm -lXft -
            lfontconfig -lfreetype -o xnedit
     97     /usr/bin/ld: file.o: in function `doOpen':
  >> 98     file.c:(.text+0x1660): undefined reference to `libiconv_open'
  >> 99     /usr/bin/ld: file.c:(.text+0x167e): undefined reference to `libiconv'
  >> 100    /usr/bin/ld: file.c:(.text+0x1f9c): undefined reference to `libiconv_close'
```
What I did there was what it took to get working, if there is a better way please tell me, I am in the dark here a little.